bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/PWAImpl.java

```diff
@@ -0,0 +1,137 @@
+
+import com.adobe.cq.wcm.core.components.models.PWA;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+
+@Model(adaptables = Resource.class,
+    adapters = {PWA.class})
+public class PWAImpl implements PWA {
+
+    static final String CONTENT_PATH = "/content/";
+
+    private boolean isEnabled = false;
+    private String manifestPath = "";
+    private String serviceWorkerPath = "";
+    private String themeColor = "";
+    private String iconPath = "";
+
+    @Self
+    private Resource resource;
+
+    @PostConstruct
+    protected void initModel() {
+        ValueMap valueMap = resource.getValueMap();
+        Boolean isPWAEnabled = valueMap.get(PN_ENABLE_PWA, Boolean.class);
+        this.isEnabled = (isPWAEnabled != null) ? isPWAEnabled : false;
+        if (!this.isEnabled) {
+            return;
+        }
+
+        this.themeColor = colorToHex(valueMap.get(PN_THEME_COLOR, ""));
+        this.iconPath = valueMap.get(PN_PWA_ICON, "");
+
+        String startURL = valueMap.get(PN_START_URL, "");
+        this.manifestPath = replaceSuffix(startURL, MANIFEST_NAME);
+
+        PageManager pageManager = resource.getResourceResolver().adaptTo(PageManager.class);
+        if (pageManager != null) {
+            Page page = pageManager.getContainingPage(resource);
+            if (page != null) {
+                String mappingName = page.getPath().replace("/", ".").substring(CONTENT_PATH.length());
+                this.serviceWorkerPath = "/" + mappingName + "sw.js";
+            }
+        }
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return this.isEnabled;
+    }
+
+    @Override
+    public String getThemeColor() {
+        return this.themeColor;
+    }
+
+    @Override
+    public String getIconPath() {
+        return this.iconPath;
+    }
+
+    @Override
+    public String getManifestPath() {
+        return this.manifestPath;
+    }
+
+    @Override
+    public String getServiceWorkerPath() {
+        return this.serviceWorkerPath;
+    }
+
+    private String colorToHex(String color) {
+        if (color == null || color.length() == 0) {
+            return "";
+        }
+
+        if (color.startsWith("#")) {
+            return color;
+        }
+
+        if (!color.startsWith("rgb")) {
+            return "";
+        }
+
+        try {
+            String[] parts = color.split(",");
+            String r = Integer.toHexString(Integer.parseInt(parts[0].substring(parts[0].indexOf("(") + 1)));
+            String g = Integer.toHexString(Integer.parseInt(parts[1]));
+            String b;
+
+            if (color.startsWith("rgba")) {
+                b = Integer.toHexString(Integer.parseInt(parts[2]));
+            }
+            else {
+                b = Integer.toHexString(Integer.parseInt(parts[2].substring(0, parts[2].indexOf(")"))));
+            }
+
+            return "#" + (r.length() == 2 ? r : "0" + r) + (g.length() == 2 ? g : "0" + g) + (b.length() == 2 ? b : "0" + b);
```

Comment:
I'd use a regex to get the RGB values (more reliable than direct string manipulation), see e.g. https://stackoverflow.com/questions/7613996/parsing-a-rgb-x-x-x-string-into-a-color-object

Reply:
Done.